docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,159 @@
+
+Volume plugins can implement any applicable interfaces.  Each plugin will document its own support for dynamic provisioning.
+
+
+### Goals
+
+* Allow administrators to describe minimum and maximum storage levels comprised of many kinds of PersistentVolumes
+* Allow the dynamic creation and reclamation of persistent volumes (to the fullest extent for each type of storage provider)
+
+
+#### API types
+
+```
+// PersistentVolumeController represents the configuration of a persistent volume.
+type PersistentVolumeController struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired specification of this PV controller.
+	Spec PersistentVolumeControllerSpec `json:"spec,omitempty"`
+
+	// Status is the current status of this PV controller.
+	Status PersistentVolumeControllerStatus `json:"status,omitempty"`
+}
+
+// PersistentVolumeControllerSpec is the specification of a PV controller.
+// The controller creates volumes exactly like the Template until the unbound count equals the replica count
+type PersistentVolumeControllerSpec struct {
+	// MinimumReplicas is the minimum number of unbound persistent volumes of this type desired in the system maintain
+	MinimumReplicas int `json:"minimumReplicas"`
+
+	// MaximumReplicas is the maximum total number of persistent volumes desired in the system
+	MaximumReplicas int `json:"maximumReplicas"`
+	
+	// Selector is a label query over persistent volumes which are managed by this controller
+	Selector map[string]string `json:"selector"`
+
+	// Template is the description of a PersistentVolume to create new replicas from
+	Template *PersistentVolumeTemplateSpec `json:"template,omitempty"`
+}
+
+// PersistentVolumeControllerStatus represents the current status of a PVController
+type PersistentVolumeControllerStatus struct {
+	// BoundReplicas is the number of replicas of this volume that are currently bound to PersistentVolumeClaims
+	BoundReplicas int `json:"boundReplicas"`
+	UnboundReplicas int `json:"unboundReplicas"`
+}
+
+// PersistentVolumeTemplateSpec describes the persistent volume created by this controller
+type PersistentVolumeTemplateSpec struct {
+	// Metadata of the PVs created from this template.
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines a persistent volume.
+	Spec PersistentVolumeSpec `json:"spec,omitempty"`
+}
+
+```
+
+Dynamic pools of storage (e.g, GCE) will maintain `MinimumReplicas` of a `PersistentVolume` at a given time where each PVs status phase is ` Available`, up to `MaximumReplicas`.  Storage levels will increase by one until the minimum or maximum levels are reached.  Recycling a PV means the current number of available volumes might exceed the minimum replica count.  Plugins implementing the Deleter interface will delete PVs where the `PersistentVolumeReclaimPolicy` is set to Delete.  The default reclaim policy is Retain.
```

Comment:
Yeah, and I missed the other reference.  I changed this doc in the proposal PR to make this clearer.

https://github.com/kubernetes/kubernetes/pull/6773